src/main/java/org/sonarlint/intellij/util/SonarLintUtils.java

```diff
@@ -231,11 +234,30 @@ public static void configureProxy(String host, ServerConfiguration.Builder build
       builder.proxy(proxy);
 
       if (httpConfigurable.PROXY_AUTHENTICATION) {
-        builder.proxyCredentials(httpConfigurable.PROXY_LOGIN, httpConfigurable.getPlainProxyPassword());
+        // Different ways to fetch login based on runtime version (SLI-95)
+        try {
+          Object proxyLogin = tryGetProxyLogin(httpConfigurable);
+          if(proxyLogin != null) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS134)